src/llmq/quorums_instantsend.h

```diff
@@ -52,23 +52,100 @@ class CInstantSendDb
+     * @param hash The hash of the InstantSend Lock
+     * @param islock The InstantSend Lock object itself
+     */
     void WriteNewInstantSendLock(const uint256& hash, const CInstantSendLock& islock);
+    /**
+     * This method removes a InstantSend Lock from instantsend.dat and is called when a tx with an IS lock is confirmed and Chainlocked
+     * @param batch Object used to batch many calls together
+     * @param hash The hash of the InstantSend Lock
+     * @param islock The InstantSend Lock object itself
+     */
     void RemoveInstantSendLock(CDBBatch& batch, const uint256& hash, CInstantSendLockPtr islock);
 
+    /**
+     * This method updates a DB entry for an InstantSend Lock from being not included in a block to being included in a block
+     * @param hash The hash of the InstantSend Lock
+     * @param nHeight The height that the transaction was included at
+     */
     void WriteInstantSendLockMined(const uint256& hash, int nHeight);
+    /**
+     * This method is called when a block gets orphaned for every tx in that block. It marks the ISLocked tx as not included in a block
+     * @param hash The hash of the InstantSend Lock
+     * @param nHeight The height of the block being orphaned
+     */
     void RemoveInstantSendLockMined(const uint256& hash, int nHeight);
+    /**
+     * Marks an InstantSend Lock as archived.
+     * @param batch Object used to batch many calls together
+     * @param hash The hash of the InstantSend Lock
+     * @param nHeight The height that the transaction was included at
+     */
     void WriteInstantSendLockArchived(CDBBatch& batch, const uint256& hash, int nHeight);
+    /**
+     * Archives and deletes all IS Locks which were mined into a block before nUntilHeight
+     * @param nUntilHeight Removes all IS Locks confirmed up until nUntilHeight
+     * @return returns an unordered_map of the hash of the IS Locks and a pointer object to the IS Locks for all IS Locks which were removed
+     */
     std::unordered_map<uint256, CInstantSendLockPtr> RemoveConfirmedInstantSendLocks(int nUntilHeight);
+    /**
+     * Removes IS Locks from the archive if the tx was confirmed 100 blocks before nUntilHeight
+     * @param nUntilHeight the height from which to base the remove of archive IS Locks
+     */
     void RemoveArchivedInstantSendLocks(int nUntilHeight);
+    /**
+     * Checks if the archive contains a certain IS Lock Hash
+     * @param islockHash The hash to check for
+     * @return Returns true if that IS Lock hash is present in the archive, otherwise returns false
+     */
     bool HasArchivedInstantSendLock(const uint256& islockHash);
+    /**
+     * Gets the number of IS Locks which have not been confirmed by a block
+     * @return size_t value of the number of IS Locks not confirmed by a block
+     */
     size_t GetInstantSendLockCount();
 
+    /**
+     * Gets a pointer to the IS Lock based on the hash
```

Comment:
"pointer to the IS lock" sounds like there is only one IS lock object for that hash, which is not true. This (and other) methods might return a new object each time called, but this also can't be guaranteed (due to the use of a cache). Maybe add a note about object ownership and that callers are not allowed to modify the object?